contracts/0.4.24/Lido.sol

```diff
@@ -355,323 +436,301 @@ contract Lido is ILido, StETH, AragonApp {
+        return IStakingRouter(getLidoLocator().stakingRouter()).getWithdrawalCredentials();
     }
 
     /**
-    * @dev Sets limit on amount of ETH to withdraw from execution layer rewards vault per LidoOracle report
-    * @param _limitPoints limit in basis points to amount of ETH to withdraw per LidoOracle report
-    */
-    function setELRewardsWithdrawalLimit(uint16 _limitPoints) external {
-        _auth(SET_EL_REWARDS_WITHDRAWAL_LIMIT_ROLE);
-
-        _setBPValue(EL_REWARDS_WITHDRAWAL_LIMIT_POSITION, _limitPoints);
-        emit ELRewardsWithdrawalLimitSet(_limitPoints);
+     * @notice Returns legacy oracle
+     * @dev DEPRECATED: the `AccountingOracle` superseeded the old one
+     */
+    function getOracle() external view returns (address) {
+        return getLidoLocator().legacyOracle();
     }
 
     /**
-    * @notice Updates beacon stats, collects rewards from LidoExecutionLayerRewardsVault and distributes all rewards if beacon balance increased
-    * @dev periodically called by the Oracle contract
-    * @param _beaconValidators number of Lido's keys in the beacon state
-    * @param _beaconBalance summarized balance of Lido-controlled keys in wei
-    */
-    function handleOracleReport(uint256 _beaconValidators, uint256 _beaconBalance) external whenNotStopped {
-        require(msg.sender == getOracle(), "APP_AUTH_FAILED");
+     * @notice Returns the treasury address
+     * @dev DEPRECATED: use LidoLocator.treasury()
+     */
+    function getTreasury() external view returns (address) {
+        return getLidoLocator().treasury();
+    }
+
 
+    /// @dev updates Consensus Layer state according to the current report
+    function _processClStateUpdate(
+        uint256 _postClValidators,
+        uint256 _postClBalance
+    ) internal returns (int256 clBalanceDiff) {
         uint256 depositedValidators = DEPOSITED_VALIDATORS_POSITION.getStorageUint256();
-        require(_beaconValidators <= depositedValidators, "REPORTED_MORE_DEPOSITED");
+        require(_postClValidators <= depositedValidators, "REPORTED_MORE_DEPOSITED");
 
-        uint256 beaconValidators = BEACON_VALIDATORS_POSITION.getStorageUint256();
-        // Since the calculation of funds in the ingress queue is based on the number of validators
-        // that are in a transient state (deposited but not seen on beacon yet), we can't decrease the previously
-        // reported number (we'll be unable to figure out who is in the queue and count them).
-        // See LIP-1 for details https://github.com/lidofinance/lido-improvement-proposals/blob/develop/LIPS/lip-1.md
-        require(_beaconValidators >= beaconValidators, "REPORTED_LESS_VALIDATORS");
-        uint256 appearedValidators = _beaconValidators.sub(beaconValidators);
+        uint256 preClValidators = CL_VALIDATORS_POSITION.getStorageUint256();
+        require(_postClValidators >= preClValidators, "REPORTED_LESS_VALIDATORS");
```

Comment:
NB
depositedValidators are total amount of ever deposited validators
_postClValidators  are total amount of ever deposited validators

withdrawals would no remove validators from the state, just changing the status

Reply:
done